ateam_kenobi/src/skills/line_kick.hpp

```diff
@@ -24,58 +24,144 @@
 
 #include <ateam_msgs/msg/robot_motion_command.hpp>
 #include <ateam_common/robot_constants.hpp>
-#include "stp/skill.hpp"
+#include "kick_skill.hpp"
 #include "types/world.hpp"
 #include "play_helpers/easy_move_to.hpp"
 
 namespace ateam_kenobi::skills
 {
 
-class LineKick : public stp::Skill
+class LineKick : public KickSkill
 {
 public:
-  explicit LineKick(stp::Options stp_options);
+  explicit LineKick(
+    stp::Options stp_options,
+    KickSkill::WaitType wait_type = KickSkill::WaitType::KickWhenReady);
 
-  void setTargetPoint(ateam_geometry::Point point)
+  void Reset() override
   {
-    target_point_ = point;
+    KickSkill::Reset();
+    state_ = State::MoveBehindBall;
   }
 
-  ateam_geometry::Point getAssignmentPoint(const World & world);
+  void SetTargetPoint(ateam_geometry::Point point)
+  {
+    if (state_ != State::KickBall && state_ != State::FaceBall) {
+      target_point_ = point;
+    }
+  }
+
+  ateam_geometry::Point GetAssignmentPoint(const World & world);
+
+  ateam_msgs::msg::RobotMotionCommand RunFrame(const World & world, const Robot & robot);
+
+  bool IsDone()
+  {
+    return state_ == State::Done;
+  }
 
-  void setKickSpeed(double speed)
+  /**
+   * @brief Set the default obstacles planner option on the internal EasyMoveTo
+   */
+  void SetUseDefaultObstacles(bool use_obstacles)
   {
-    kick_speed_ = speed;
+    path_planning::PlannerOptions options = easy_move_to_.getPlannerOptions();
+    options.use_default_obstacles = use_obstacles;
+    easy_move_to_.setPlannerOptions(options);
   }
 
-  ateam_msgs::msg::RobotMotionCommand runFrame(const World & world, const Robot & robot);
+  double move_to_ball_velocity = 1.8;
+  double robot_perp_dist_to_ball_threshold = 0.008;
+  double angle_threshold = 0.1;
+  double kick_drive_velocity = 0.4;
+
+  // Allow the robot to push up against obstacles / other robots (slowly)
+  bool cowabunga = false;
```

Comment:
Are we keeping cowabunga mode or nah

Reply:
I think we'll keep it for now. As we fix up motion control and come back to kick skills, we'll do some more experiments to see if cowabunga mode is useful.